cqf-fhir-cr/src/main/java/org/opencds/cqf/fhir/cr/measure/r4/R4CareGapsBundleBuilder.java

```diff
@@ -0,0 +1,322 @@
+ */
+public class R4CareGapsBundleBuilder {
+    private static final Map<String, CodeableConceptSettings> CARE_GAPS_CODES = ImmutableMap.of(
+            "http://loinc.org/96315-7",
+            new CodeableConceptSettings().add("http://loinc.org", "96315-7", "Gaps in care report"),
+            "http://terminology.hl7.org/CodeSystem/v3-ActCode/CAREGAP",
+            new CodeableConceptSettings()
+                    .add("http://terminology.hl7.org/CodeSystem/v3-ActCode", "CAREGAP", "Care Gaps"));
+
+    private final Repository repository;
+
+    private final MeasureEvaluationOptions measureEvaluationOptions;
+    protected final Map<String, Resource> configuredResources;
+
+    private static final FhirContext fhirContext = FhirContext.forCached(FhirVersionEnum.R4);
+
+    private CareGapsProperties careGapsProperties;
+
+    private String serverBase;
+
+    private final R4MeasureServiceUtils r4MeasureServiceUtils;
+
+    private final R4MultiMeasureService r4MultiMeasureService;
+
+    public R4CareGapsBundleBuilder(
+            CareGapsProperties careGapsProperties,
+            Repository repository,
+            MeasureEvaluationOptions measureEvaluationOptions,
+            String serverBase,
+            Map<String, Resource> configuredResources) {
+        this.repository = repository;
+        this.careGapsProperties = careGapsProperties;
+        this.measureEvaluationOptions = measureEvaluationOptions;
+        this.serverBase = serverBase;
+        this.configuredResources = configuredResources;
+
+        r4MeasureServiceUtils = new R4MeasureServiceUtils(repository);
+        r4MultiMeasureService = new R4MultiMeasureService(repository, measureEvaluationOptions, serverBase);
+    }
+
+    public List<Parameters.ParametersParameterComponent> makePatientBundles(
+            IPrimitiveType<Date> periodStart,
+            IPrimitiveType<Date> periodEnd,
+            List<String> subjects,
+            List<String> statuses,
+            List<IdType> measureIds) {
+
+        // retrieve reporter from configuration
+        String reporter = RESOURCE_TYPE_ORGANIZATION.concat("/" + careGapsProperties.getCareGapsReporter());
+
+        List<ParametersParameterComponent> paramResults = new ArrayList<>();
+
+        for (String subject : subjects) {
+            // Measure Reports
+            Bundle result = r4MultiMeasureService.evaluate(
+                    measureIds,
+                    null,
+                    null,
+                    periodStart.getValueAsString(),
+                    periodEnd.getValueAsString(),
+                    MeasureEvalType.SUBJECT.toCode(),
+                    subject,
+                    null,
+                    null,
+                    null,
+                    null,
+                    null,
+                    null,
+                    reporter);
+
+            // Patient
+            var patient = repository.read(Patient.class, new IdType(subject));
```

Comment:
Please pass in resource type to the 2 arg IdType constructor

Reply:
"subject" comes in as ResourceType/{Id}, so adding in IdType(ResourceType, subject) results in "Patient/Patient/{id}"